src/Catalyst.Core.Lib/Util/Duration.cs

```diff
@@ -0,0 +1,201 @@
+
+                while (sr.Peek() != -1)
+                {
+                    result += ParseComponent(sr);
+                }
+            }
+
+            if (negative)
+                return TimeSpan.FromTicks(-result.Ticks);
+
+            return result;
+        }
+
+        private static TimeSpan ParseComponent(StringReader reader)
+        {
+            var value = ParseNumber(reader);
+            var unit = ParseUnit(reader);
+
+            switch (unit)
+            {
+                case "h":
+                    return TimeSpan.FromHours(value);
+                case "m":
+                    return TimeSpan.FromMinutes(value);
+                case "s":
+                    return TimeSpan.FromSeconds(value);
+                case "ms":
+                    return TimeSpan.FromMilliseconds(value);
+                case "us":
+                case "µs":
+                    return TimeSpan.FromTicks((long) (value * TicksPerMicrosecond));
+                case "ns":
+                    return TimeSpan.FromTicks((long) (value * TicksPerNanosecond));
+                case "":
+                    throw new FormatException("Missing IPFS duration unit.");
+                default:
+                    throw new FormatException($"Unknown IPFS duration unit '{unit}'.");
+            }
+        }
+
+        private static double ParseNumber(StringReader reader)
+        {
+            var s = new StringBuilder();
+            while (true)
+            {
+                var c = (char) reader.Peek();
+                if (char.IsDigit(c) || c == '.')
+                {
+                    s.Append(c);
+                    reader.Read();
+                    continue;
+                }
+
+                return double.Parse(s.ToString(), CultureInfo.InvariantCulture);
+            }
+        }
+
+        private static string ParseUnit(StringReader reader)
+        {
+            var s = new StringBuilder();
+            while (true)
+            {
+                var c = (char) reader.Peek();
+                if (char.IsDigit(c) || c == '.' || c == (char) 0xFFFF)
+                    break;
+                s.Append(c);
+                reader.Read();
+            }
+
+            return s.ToString();
+        }
+
+        /// <summary>
+        ///   Converts the <see cref="TimeSpan"/> to the equivalent string representation of an 
+        ///   IPFS duration.
+        /// </summary>
+        /// <param name="duration">
+        ///   The <see cref="TimeSpan"/> to convert.
+        /// </param>
+        /// <param name="zeroValue">
+        ///   The string representation, when the <paramref name="duration"/> 
+        ///   is equal to <see cref="TimeSpan.Zero"/>/
+        /// </param>
+        /// <returns>
+        ///   The IPFS duration string representation.
+        /// </returns>
+        public static string Stringify(TimeSpan duration, string zeroValue = "0s")
+        {
+            if (duration.Ticks == 0)
```

Comment:
![Codacy](https://app.codacy.com/assets/images/favicon.png) Issue found: [Add curly braces around the nested statement(s) in this 'if' block.](https://app.codacy.com/manual/catalyst-network/Catalyst.Framework/pullRequest?prid=4832783)